### PR TITLE
Clean up offset storage contract usage

### PR DESCRIPTION
**Pull request by ladytrekker**
## Description

A minor refactoring PR which includes:
- remove doubled abi for KlimaRetirementStorage
- create new lib helper for creating this contract
- use this contract in App and Site
- use types
- align namings for types
- align formatting of data in both App & Site
- replace method `retirements` with `getRetirementTotals` as both are returning the same data
- **Additional Fix:** Mapping of children with missing key in component CarbonTonnesBreakdownCard

## Test

**APP:**
- Open the preview link for the App: https://klimadao-app-git-clean-up-offset-klimadao.vercel.app/#/offset
- Connect your wallet
- Check in the boxes at the top right that the total numbers of retirements are rendered

**SITE:**
- Open the preview link for the App: https://klimadao-site-git-clean-up-offset-klimadao.vercel.app/
- add to the URL `<YOU-WALLET-ADDRESS>`
- => `https://klimadao-site-git-clean-up-offset-klimadao.vercel.app/<YOU-WALLET-ADDRESS>`
- Open the page and you should see the same data as in App above (not styled yet)

## Checklist

- [x] Building the site with `npm run build-site` works without errors
- [x] Building the app with `npm run build-app` works without errors
- [x] I formatted JS and TS files with running `npm run format-all`

### CODE REVIEW

File: lib/utils/getRetirement/index.ts [hunk 1]
```diff
@@ -9,37 +9,48 @@ import {
...
 } from "../../types/offset";
 
+export const createRetirementStorageContract = (
+  provider: providers.JsonRpcProvider
+) => {
+  return new ethers.Contract(
+    addresses["mainnet"].retirementStorage,
+    KlimaRetirementStorage.abi,
+    provider
+  );
```
Comment (Atmosfearful) [inline lib/utils/getRetirement/index.ts:25]: I wonder if in the future we should just instantiate and export every contract once, with a default provider, maybe from `@klimadao/lib`
and then when we use them for user transactions which need private key, we set the provider

```ts
import { retirementStorage } from "@klimadao/lib/contracts";

const doTransaction = (params) => {
  retirementStorage.provider = params.provider;
  const totals = retirementStorage.getRetirementTotals(params.address);
}
```

**Review by Atmosfearful — APPROVED** [review]
nice cleanups ty

**Comment by ShimonD-KlimaDAO** [comment]
I tried two separate wallets, and both display a 404 error on the second testing step (the site URL). @ladytrekker , sending you a DM with the info.

**Comment by ladytrekker** [comment]
> I tried two separate wallets, and both display a 404 error on the second testing step (the site URL). @ladytrekker , sending you a DM with the info.

Thanks for testing @ShimonD-KlimaDAO 

I forgot to add "/retirement" in the description above
The correct URL is `https://klimadao-site-git-clean-up-offset-klimadao.vercel.app/retirement/<YOU-WALLET-ADDRESS>`

Updated above too for documentation reasons (if anyone every looks at a merged PR again)

File: lib/utils/getRetirement/index.ts (hunk 1 above)
Comment (ladytrekker) [reply lib/utils/getRetirement/index.ts:25]: Good idea, let´s keep that in mind.

**Pull request merged**